plugins/module_utils/network/sonic/config/mclag/mclag.py

```diff
@@ -180,19 +214,167 @@ def _state_deleted(self, want, have):
+                    del_peer_gateway_vlans = self.get_vlan_range_common(want_peer_gateway['vlans'], have['peer_gateway']['vlans'])
+
             new_have = self.remove_default_entries(have)
             d_diff = get_diff(want, new_have, TEST_KEYS, is_skeleton=True)
             diff_want = get_diff(want, d_diff, TEST_KEYS, is_skeleton=True)
+
+            if del_unique_ip_vlans:
+                diff_want['unique_ip'] = {'vlans': del_unique_ip_vlans}
+            if del_peer_gateway_vlans:
+                diff_want['peer_gateway'] = {'vlans': del_peer_gateway_vlans}
+
             if diff_want:
-                requests = self.get_delete_mclag_attribute_request(want, diff_want)
+                requests = self.get_delete_mclag_attribute_requests(want, diff_want)
                 if len(requests) > 0:
                     commands = update_states(diff_want, "deleted")
         return commands, requests
 
+    def _state_replaced_overridden(self, want, have, state):
+        """ The command generator when state is replaced/overridden
+
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = []
+        requests = []
+        if want and not have:
+            commands = [update_states(want, state)]
+            requests = self.get_create_mclag_requests(want, want)
+        elif not want and have:
+            commands = [update_states(have, 'deleted')]
+            requests = self.get_delete_all_mclag_domain_requests(have)
+        elif want and have:
+            add_command = {}
+            del_command = {}
+            delete_all = False
+
+            # If 'domain_id' is modified:
+            # 1) Delete all mclag configuration.
+            # 2) When state is replaced, reconfigure options with
+            #    values prior to deletion if they are not specified.
+            if want['domain_id'] != have['domain_id']:
+                del_command = have
+                delete_all = True
+
+                for option in self.mclag_attrs:
+                    if want.get(option) is None:
+                        if state == 'replaced' and have.get(option) is not None:
+                            add_command[option] = have[option]
+                    else:
+                        if option == 'unique_ip' and want['unique_ip'].get('vlans'):
+                            add_command['unique_ip'] = want['unique_ip']
+                        elif option == 'peer_gateway' and want['peer_gateway'].get('vlans'):
+                            add_command['peer_gateway'] = want['peer_gateway']
+                        elif option == 'members' and want['members'].get('portchannels'):
+                            add_command['members'] = want['members']
+                        else:
+                            add_command[option] = want[option]
+            else:
+                have = have.copy()
```

Comment:
What is the reason for this instruction and the following instruction?

Reply:
In the following code block, after computing the diff in VLAN range for 'unique_ip' and 'peer_gateway' those options in 'want' and 'have' are popped before computing the diff for other options using 'get_diff'. 

The copy was done to avoid modifying the actual 'want' and 'have' passed to this method.